app/services/foreman/plugin.rb

```diff
@@ -83,6 +83,12 @@ def find(id)
         registered_plugins[id.to_sym]
       end
 
+      # Evaluates additional setup steps after initial registration
+      # useful when you need to define methods later e.g. from config.to_prepare
+      def setup(id, &block)
```

Comment:
it can be called from `after_initialize` as well as from any other place (even another initializer)